daemon/graphdriver/overlay2/check.go

```diff
@@ -10,16 +10,26 @@ import (
 	"path/filepath"
 	"syscall"
 
+	"github.com/containerd/containerd/sys"
 	"github.com/docker/docker/pkg/system"
 	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
+var errRunningInUserNS = errors.New("running in a user namespace")
+
 // doesSupportNativeDiff checks whether the filesystem has a bug
 // which copies up the opaque flag when copying up an opaque
 // directory or the kernel enable CONFIG_OVERLAY_FS_REDIRECT_DIR.
 // When these exist naive diff should be used.
+//
+// When running in a user namespace, returns errRunningInUserNS
+// immediately.
 func doesSupportNativeDiff(d string) error {
+	if sys.RunningInUserNS() {
+		return errRunningInUserNS
```

Comment:
```suggestion
		return errors.New("running in a user namespace")
```

as this is only called _once_ from `useNaiveDiff`.

Reply:
fixed